crates/storage/provider/src/providers/snapshot/manager.rs

```diff
@@ -156,20 +156,19 @@ impl SnapshotProvider {
         tx_range: &RangeInclusive<u64>,
     ) -> ProviderResult<SnapshotJarProvider<'_>> {
         let key = (*block_range.end(), segment);
-        if let Some(jar) = self.map.get(&key) {
-            Ok(jar.into())
-        } else {
-            let jar = NippyJar::load(&self.path.join(segment.filename(block_range, tx_range)))
-                .map(|jar| {
+        let entry = match self.map.entry(key) {
```

Comment:
this would acquire a lock which is not desirable

Reply:
so does calling get twice